--- a/src/platform/unix.rs	2026-07-17 21:05:36
+++ b/src/platform/unix.rs	2026-07-17 21:05:36
@@ -131,6 +131,9 @@
             target_os = "solaris",
             target_os = "macos",
             target_os = "ios",
+            target_os = "tvos",
+            target_os = "watchos",
+            target_os = "visionos",
             target_os = "dragonfly",
             target_os = "freebsd",
             target_os = "openbsd",
@@ -169,6 +172,9 @@
             target_os = "solaris",
             target_os = "macos",
             target_os = "ios",
+            target_os = "tvos",
+            target_os = "watchos",
+            target_os = "visionos",
             target_os = "dragonfly",
             target_os = "freebsd",
             target_os = "openbsd",
