--- a/src/date.rs
+++ b/src/date.rs
@@ -10,7 +10,7 @@
 #[cfg(windows)]
 use chrono::{Datelike, Timelike};
 use clap::{crate_version, Arg, ArgAction, Command};
-#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios")), not(target_os = "redox")))]
+#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos")), not(target_os = "redox")))]
 use libc::{clock_settime, timespec, CLOCK_REALTIME};
 use std::fs::File;
 use std::io::{BufRead, BufReader};
@@ -63,9 +63,9 @@
  for date and time to the indicated precision.
  Example: 2006-08-14 02:34:56-06:00";
 
-#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "redox")))]
+#[cfg(not(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos", target_os = "redox")))]
 static OPT_SET_HELP_STRING: &str = "set time described by STRING";
-#[cfg(any(target_os = "macos", target_os = "ios"))]
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos"))]
 static OPT_SET_HELP_STRING: &str = "set time described by STRING (not available on mac yet)";
 #[cfg(target_os = "redox")]
 static OPT_SET_HELP_STRING: &str = "set time described by STRING (not available on redox yet)";
@@ -420,7 +420,7 @@
     unimplemented!("setting date not implemented (unsupported target)");
 }
 
-#[cfg(any(target_os = "macos", target_os = "ios"))]
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos"))]
 fn set_system_datetime(_date: DateTime<Utc>) -> UResult<()> {
     Err(USimpleError::new(
         1,
@@ -436,7 +436,7 @@
     ))
 }
 
-#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios")), not(target_os = "redox")))]
+#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios", target_os = "tvos", target_os = "watchos", target_os = "visionos")), not(target_os = "redox")))]
 /// System call to set date (unix).
 /// See here for more:
 /// `<https://doc.rust-lang.org/libc/i686-unknown-linux-gnu/libc/fn.clock_settime.html>`
